src/test/java/org/opensearch/ad/ml/ModelManagerTests.java

```diff
@@ -8,992 +8,1060 @@
-
-       AnomalyDescriptor descriptor = new AnomalyDescriptor(point, 0);
-       descriptor.setRCFScore(score);
-       descriptor.setNumberOfTrees(numTrees);
-       descriptor.setDataConfidence(confidence);
-       descriptor.setAnomalyGrade(grade);
-       descriptor.setAttribution(attributionVec);
-       descriptor.setTotalUpdates(numSamples);
-       descriptor.setRelativeIndex(relativeIndex);
-       descriptor.setRelevantAttribution(currentTimeAttribution);
-       descriptor.setPastValues(pastalues);
-       descriptor.setExpectedValuesList(expectedValuesList);
-       descriptor.setLikelihoodOfValues(likelihood);
-       descriptor.setThreshold(threshold);
-
-       when(rForest.process(any(), anyLong())).thenReturn(descriptor);
-
-       ActionListener<ThresholdingResult> listener = mock(ActionListener.class);
-       modelManager.getTRcfResult(detectorId, rcfModelId, point, listener);
-
-       ThresholdingResult expected = new ThresholdingResult(
-           grade,
-           confidence,
-           score,
-           numSamples,
-           relativeIndex,
-           currentTimeAttribution,
-           pastalues,
-           expectedValuesList,
-           likelihood,
-           threshold,
-           numTrees
-       );
-       verify(listener).onResponse(eq(expected));
-
-       descriptor.setTotalUpdates(numSamples + 1L);
-       listener = mock(ActionListener.class);
-       modelManager.getTRcfResult(detectorId, rcfModelId, point, listener);
-
-       ArgumentCaptor<ThresholdingResult> responseCaptor = ArgumentCaptor.forClass(ThresholdingResult.class);
-       verify(listener).onResponse(responseCaptor.capture());
-       assertEquals(0.091353632, responseCaptor.getValue().getConfidence(), 1e-6);
-   }
-
-   @Test
-   public void getRcfResult_throwToListener_whenNoCheckpoint() {
-       doAnswer(invocation -> {
-           ActionListener<Optional<RandomCutForest>> listener = invocation.getArgument(1);
-           listener.onResponse(Optional.empty());
-           return null;
-       }).when(checkpointDao).getTRCFModel(eq(rcfModelId), any(ActionListener.class));
-
-       ActionListener<ThresholdingResult> listener = mock(ActionListener.class);
-       modelManager.getTRcfResult(detectorId, rcfModelId, new double[0], listener);
-
-       verify(listener).onFailure(any(ResourceNotFoundException.class));
-   }
-
-   @Test
-   public void getRcfResult_throwToListener_whenHeapLimitExceed() {
-       doAnswer(invocation -> {
-           ActionListener<Optional<ThresholdedRandomCutForest>> listener = invocation.getArgument(1);
-           listener.onResponse(Optional.of(rcf));
-           return null;
-       }).when(checkpointDao).getTRCFModel(eq(rcfModelId), any(ActionListener.class));
-
-       when(jvmService.info().getMem().getHeapMax().getBytes()).thenReturn(1_000L);
-       final Set<Setting<?>> settingsSet = Stream
-           .concat(
-               ClusterSettings.BUILT_IN_CLUSTER_SETTINGS.stream(),
```

Comment:
Curious (not for this PR) what's in these built-in settings and whether they need an equivalent in SDKClusterSettings...

Reply:
That's a great catch. I missed it during copy/paste of the new pattern for ClusterSettings. I have added `BUILT_IN_CLUSTER_SETTINGS` back.